romancal/associations/load_as_asn.py

```diff
@@ -0,0 +1,207 @@
+
+    Notes
+    -----
+    This class is normally not instantiated.
+    the `load` method should be used as the factory
+    method to read an association or create one from
+    a string or `Datamodel` object, or a list of such
+    objects.
+    """
+
+    @classmethod
+    def load(
+        cls,
+        obj,
+        meta=DEFAULT_ASN_META,
+        registry=AssociationRegistry,
+        rule=Association,
+        product_name_func=None,
+    ):
+        """Load object and return an association of it
+
+        Parameters
+        ----------
+        obj : Association, str, Datamodel, [str[,...]], [Datamodel[,...]]
+            The obj to return as an association
+
+        registry : AssociationRegistry
+            The registry to use to load an association file with
+
+        rule : Association
+            The rule to use if an association needs to be created
+
+        product_name_func : func
+            A function, when given the argument of `obj`, or
+            if `obj` is a list, each item in `obj`, returns
+            a string that will be used as the product name in
+            the association.
+
+        Attributes
+        ----------
+        Along with the attributes belonging to an association, the
+        following are added:
+
+        filename : str
+            The name of the association file, if such a file
+            where passed in. Otherwise a default value is given.
+
+        Returns
+        -------
+        association : Association
+            An association created using given obj
+        """
+        try:
+            with open(obj) as fp:
+                pure_asn = load_asn(fp, registry=registry)
+        except Exception:
+            if not isinstance(obj, list):
+                obj = [obj]
+            asn = asn_from_list(
+                obj,
+                rule=rule,
+                meta=DEFAULT_ASN_META,
+                product_name_func=product_name_func,
+            )
+            asn.filename = DEFAULT_NAME
+        else:
+            asn = rule()
+            asn.update(pure_asn)
+            asn.filename = obj
+
+        return asn
+
+
+class LoadAsLevel2Asn(LoadAsAssociation):
+    """Read in or create a Level2 association"""
+
+    @classmethod
+    def load(cls, obj, basename=None):
+        """Open object and return a Level2 association of it
+
+        Parameters
+        ----------
+        obj : Association, str, Datamodel, [str[,...]], [Datamodel[,...]]
+            The obj to return as an association
+
+        basename : str
+            If specified, use as the basename, with an index appended.
+
+        Attributes
+        ----------
+        Along with the attributes belonging to a Level2 association, the
+        following are added:
+
+        filename : str
+            The name of the association file, if such a file
+            where passed in. Otherwise a default value is given.
+
+        Returns
+        -------
+        association : DMSLevel2bBase
```

Comment:
Should we get away from JWST names and levels, since there's noLevel 2b for Roman?

Reply:
That's fine. I did this for expediency and when the associations were much more poorly defined. Can this be a separate ticket to clean this up?